embedded/store/key_reader.go

```diff
@@ -24,23 +24,27 @@ import (
 )
 
 type Snapshot struct {
-	st   *ImmuStore
-	snap *tbtree.Snapshot
+	st             *ImmuStore
+	snap           *tbtree.Snapshot
+	refInterceptor valueRefInterceptor
 }
 
-type FilterFn func(valRef *ValueRef) bool
+type valueRefInterceptor func(key []byte, valRef ValueRef) ValueRef
+
+type FilterFn func(valRef ValueRef) bool
 
 var (
-	IgnoreDeleted FilterFn = func(valRef *ValueRef) bool {
-		return valRef.kvmd == nil || !valRef.kvmd.deleted
+	IgnoreDeleted FilterFn = func(valRef ValueRef) bool {
+		return valRef.KVMetadata() == nil || !valRef.KVMetadata().deleted
```

Comment:
Shouldn't it be:
```suggestion
		return valRef.KVMetadata() != nil && !valRef.KVMetadata().deleted
```
?

What does it mean that `KVMetadata() == nil` ? I assumed that's for old-style entries.

Reply:
entries when filter function evaluates to `false` are currently filtered out, so I'll revert so to skip when filter evaluates to true. And in such case the condition will be as you describe.
KVMetadata may be nil even in new entries, because they are optional.